File: backend/LibraryApp.Database/Entities/Book.cs
namespace LibraryApp.Database.Entities;

public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Isbn { get; set; } = string.Empty;
    public DateOnly PublishedDate { get; set; }
    public string Owner { get; set; } = string.Empty;
    public bool IsAvailable { get; set; } = true;
}

File: frontend/src/types/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  isbn: string;
  publishedDate: string;
  owner: string;
  isAvailable: boolean;
}

export interface CreateBookRequest {
  title: string;
  author: string;
  isbn: string;
  publishedDate: string;
  owner: string;
}

export interface UpdateBookRequest extends CreateBookRequest {
  isAvailable: boolean;
}
